src/modules/strings.wren

```diff
@@ -1,23 +1,17 @@
 #module=essentials
 class Strings {
-    static upcase(s) {
-        return s.bytes.map { |x|
-            if ((LOWERCASE_A..LOWERCASE_Z).contains(x)) x = x - 32
-            return String.fromByte(x)
-        }.join("")
-    }
+    static upcase(s) {upcase_(s.toString)}
+    foreign static upcase_(s)
+
+    static downcase(s) {downcase_(s.toString)}
```

Comment:
Think we probably need the same string guards here that I just pointed out on JSON?

Reply:
Did we decide against `f_upcase_` I thought I was willing to give it a shot (at least in essentials)?

Reply:
I think for JSON it makes sence to fail if you pass a non string to parse it.

For this is always a string is nothing more that convert each character to another codepoint.
If you pass a `null` to upcase it will return `"NULL"`.

about `f_upcase_` i think since  many other already have the underscore at the end, I prefer to follow that convention :)

Reply:
> If you pass a null to upcase it will return "NULL".

_Silently processing null (or other non-strings) would be a bug IMHO._  The expected/required input type is string.  Also this could be hugely expensive if you had a large data object that was accidentally passed and `toString` was called on it.  I strongly feel this is an anti-pattern (input validation via auto typecasting).  The moment a non-string is passed the program should blow up, alerting the programmer to fix the bug.

Reply:
Agree, 
ok so we wait a little until a proper assertion is in place?.
Or for this is better to use the C lang assertions until a Wren assertion is available?
Or just an 

```js
if (!(value is String)) {
  Fiber.abort("Not a string")
}
```

Reply:
In my opinion is best that we have this kind of checks in C side.
With some helpful macros to ease.

Assert params count
and param type